201-vm-win-iis-app-ssl/azuredeploy.json

```diff
@@ -0,0 +1,409 @@
+      "tags": {
+        "displayName": "appStorage"
+      },
+      "properties": {
+        "accountType": "[parameters('appStorageType')]"
+      }
+    },
+    {
+      "name": "[variables('appVMNicName')]",
+      "type": "Microsoft.Network/networkInterfaces",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2015-06-15",
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/virtualNetworks', 'appVnet')]",
+        "[resourceId('Microsoft.Network/publicIPAddresses', variables('appPublicIPName'))]"
+      ],
+      "tags": {
+        "displayName": "appVMNic"
+      },
+      "properties": {
+        "ipConfigurations": [
+          {
+            "name": "ipconfig1",
+            "properties": {
+              "privateIPAllocationMethod": "Dynamic",
+              "subnet": {
+                "id": "[variables('appVMSubnetRef')]"
+              },
+              "publicIPAddress": {
+                "id": "[resourceId('Microsoft.Network/publicIPAddresses', variables('appPublicIPName'))]"
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "name": "[parameters('appVMName')]",
+      "type": "Microsoft.Compute/virtualMachines",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2015-06-15",
+      "dependsOn": [
+        "[resourceId('Microsoft.Storage/storageAccounts', variables('appStorageName'))]",
+        "[resourceId('Microsoft.Network/networkInterfaces', variables('appVMNicName'))]"
+      ],
+      "tags": {
+        "displayName": "appVM"
+      },
+      "properties": {
+        "hardwareProfile": {
+          "vmSize": "[variables('appVMVmSize')]"
+        },
+        "osProfile": {
+          "computerName": "[parameters('appVMName')]",
+          "adminUsername": "[parameters('appVMAdminUsername')]",
+          "adminPassword": "[parameters('appVMAdminPassword')]",
+          "secrets": [
+            {              
+              "sourceVault": {
+                "id": "[resourceId(parameters('vaultResourceGroup'), 'Microsoft.KeyVault/vaults', parameters('vaultName'))]"               
+              },
+              "vaultCertificates": [
+                {
+                  "certificateUrl": "[parameters('httpssecretUrlWithVersion')]",
+                  "certificateStore": "[parameters('certificateStore')]"
+                }
+              ]
+            }
+          ]
+        },
+        "storageProfile": {
+          "imageReference": {
+            "publisher": "[variables('appVMImagePublisher')]",
+            "offer": "[variables('appVMImageOffer')]",
+            "sku": "[parameters('appVMWindowsOSVersion')]",
+            "version": "latest"
+          },
+          "osDisk": {
+            "name": "appVMOSDisk",
+            "vhd": {
+              "uri": "[concat('https://', variables('appStorageName'), '.blob.core.windows.net/', variables('appVMStorageAccountContainerName'), '/', variables('appVMOSDiskName'), '.vhd')]"
```

Comment:
Please see #1 in the best practice checklist: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md